tests/test_tacacs_server_group.rb

```diff
@@ -17,98 +14,447 @@
+  end
+
+  def destroy_aaa_group(group_name, server)
+    config("no aaa group server #{server} #{group_name}")
+  end
+
+  def create_vrf(group_name, server, vrf_name)
+    config("aaa group server #{server} #{group_name} ; use-vrf #{vrf_name}")
+  end
+
+  def create_deadtime(group_name, server, deadtime)
+    config("aaa group server #{server} #{group_name} ; deadtime #{deadtime}")
+  end
+
+  def create_source_interface(group_name, server, interface)
+    config("aaa group server #{server} #{group_name} ; " \
+           "source-interface #{interface}")
+  end
+
+  def test_create_invalid_name_tacacs
+    assert_raises(TypeError) do
+      TacacsServerGroup.new(nil)
+    end
+  end
+
+  def test_create_valid_tacacs
+    group_name = 'Group1'
+    aaa_group = TacacsServerGroup.new(group_name)
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /Group1/)
+
+    detach_aaaservergroup(aaa_group)
+  end
+
+  def test_create_valid_multiple_tacacs
+    group_name1 = 'Group1'
+    group_name2 = 'Group2'
+    aaa_group1 = TacacsServerGroup.new(group_name1)
+    aaa_group2 = TacacsServerGroup.new(group_name2)
+
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /Group1/)
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /Group2/)
+
+    detach_aaaservergroup(aaa_group1)
+    detach_aaaservergroup(aaa_group2)
+  end
+
+  def test_collection_empty_tacacs
+    clean_tacacs_config
+    aaa_group_list = TacacsServerGroup.groups
+    assert_empty(aaa_group_list,
+                 'Error: TacacsServerGroup collection is not empty')
+  end
+
+  def test_collection_single_tacacs
+    clean_tacacs_config
+    group_name1 = 'Group1'
+    group_name2 = 'Group2'
+    group_name3 = 'Group3'
+    aaa_group1 = TacacsServerGroup.new(group_name1)
+
+    groups = TacacsServerGroup.groups
+    refute_empty(groups, 'Error: TacacsServerGroup collection is not filled')
+    assert_equal(1, groups.size,
+                 'Error: TacacsServerGroup collection reporting incorrect size')
+    assert(groups.key?(group_name1),
+           "Error: TacacsServerGroup collection does contain #{group_name1}")
+    detach_aaaservergroup(aaa_group1)
+
+    create_aaa_group(group_name2, 'tacacs+')
+    create_aaa_group(group_name3, 'tacacs+')
+    groups = TacacsServerGroup.groups
+    refute_empty(groups, 'Error: TacacsServerGroup collection is not filled')
+    assert_equal(2, groups.size,
+                 'Error: TacacsServerGroup collection reporting incorrect size')
+    assert(groups.key?(group_name2),
+           "Error: TacacsServerGroup collection does contain #{group_name2}")
+    assert(groups.key?(group_name3),
+           "Error: TacacsServerGroup collection does contain #{group_name3}")
+
+    destroy_aaa_group(group_name2, 'tacacs+')
+    destroy_aaa_group(group_name3, 'tacacs+')
+  end
+
+  def test_collection_multi_tacacs
```

Comment:
Is there something subtle here? What does this test offer that isn't testing in the previous test?